### ISSUE TITLE
Performance gap between libdevice.exp and tl.exp

### ISSUE THREAD

**Issue by NaOHCC** [intel/intel-xpu-backend-for-triton#8002]
## Background

We are currently investigating a significant performance gap between Triton kernels generated by Helion and the corresponding PyTorch eager SYCL kernels for softmax on Intel XPU.

## Describe the issue

In the following softmax kernel, replacing `libdevice.exp` with `tl.exp` improves the runtime from approximately `0.1487 ms` to `0.1261 ms`, which is about a 15.2% speedup.

```python
from __future__ import annotations

import torch
import triton
import triton.language as tl

# from torch._inductor.runtime.triton_compat import libdevice
from triton.language.extra import libdevice

_BLOCK_SIZE_0 = tl.constexpr(1)
_BLOCK_SIZE_1 = tl.constexpr(1024)


@triton.jit
def __softmax_two_pass(x, out):
    pid_0 = tl.program_id(0)
    offset_0 = pid_0
    indices_0 = offset_0 + tl.zeros([1], tl.int32)
    mi = tl.full([_BLOCK_SIZE_0], float("-inf"), tl.float32)
    di = tl.full([_BLOCK_SIZE_0], 0.0, tl.float32)
    for offset_2 in tl.range(0, 4096, _BLOCK_SIZE_1):
        indices_2 = offset_2 + tl.arange(0, _BLOCK_SIZE_1).to(tl.int32)
        mi_copy = mi
        di_copy = di
        mi_copy_0 = mi_copy
        di_copy_0 = di_copy
        values = tl.load(x + (indices_0[:, None] * 4096 + indices_2[None, :] * 1), None, eviction_policy="evict_first")
        local_amax = tl.cast(tl.max(values, 1), tl.float16)
        v_0 = tl.cast(local_amax, tl.float32)
        v_1 = tl.maximum(mi_copy_0, v_0, tl.PropagateNan.ALL)
        v_2 = mi_copy_0 - v_1
        v_3 = libdevice.exp(v_2)
        v_4 = di_copy_0 * v_3
        subscript = v_1[:, None]
        v_5 = tl.cast(values, tl.float32)
        v_6 = v_5 - subscript
        v_7 = libdevice.exp(v_6)
        sum_1 = tl.cast(tl.sum(v_7, 1), tl.float32)
        di = v_4 + sum_1
        mi = v_1
    for offset_2 in tl.range(0, 4096, _BLOCK_SIZE_1):
        indices_2 = offset_2 + tl.arange(0, _BLOCK_SIZE_1).to(tl.int32)
        mi_copy_1 = mi
        di_copy_1 = di
        mi_copy_1_0 = mi_copy_1
        di_copy_1_0 = di_copy_1
        values_1 = tl.load(x + (indices_0[:, None] * 4096 + indices_2[None, :] * 1), None, eviction_policy="evict_first")
        subscript_1 = mi_copy_1_0[:, None]
        v_9 = tl.cast(values_1, tl.float32)
        v_10 = v_9 - subscript_1
        v_11 = libdevice.exp(v_10)
        subscript_2 = di_copy_1_0[:, None]
        v_12 = v_11 / subscript_2
        v_13 = tl.cast(v_12, tl.float16)
        tl.store(out + (indices_0[:, None] * 4096 + indices_2[None, :] * 1), v_13, None)


def softmax_two_pass(x: torch.Tensor):
    m, n = x.size()
    out = torch.empty_like(x)
    __softmax_two_pass[(m,)](x, out)
    return out


def call():
    torch.manual_seed(0)
    x = torch.rand((4096, 4096), dtype=torch.float16, device="xpu:0")
    out = softmax_two_pass(x)
    ms = triton.testing.do_bench(lambda: softmax_two_pass(x))
    print(f"Softmax benchmark: {ms} ms")
    return out


if __name__ == "__main__":
    call()
```

## Analysis

Intel Triton currently lowers `libdevice.exp` for `fp32` to `__imf_expf`.

`__imf_expf` is an IMF function that uses a relatively high-precision algorithm. Its behavior and performance characteristics differ from both `sycl::exp(float)` and `tl.exp`.

- `sycl::exp(float)` is forwarded to the IGC built-in function `__spirv_ocl_exp`, whose implementation is available here:
  https://github.com/intel/intel-graphics-compiler/blob/master/IGC/BiFModule/Implementation/Math/exp.cl#L17-L57
- `tl.exp` use an `exp2`-based implementation for `fp32`.
- On CUDA, both `expf(float)` and `libdevice.exp`, which is based on the `__nv_expf` intrinsic, use an `exp2`-based algorithm: https://godbolt.org/z/ox6xP3b55

## Expected Behavior

Intel Triton `libdevice.exp` should use an implementation consistent with `tl.exp` and `sycl::exp(float)` for `fp32` computations.

## Environment details

- PyTorch: `2.14.0+xpu`
- Triton-XPU: `3.8.0+git1e2d42a0`
- GPU: Intel B70
